crypto/dilithium/ml_dsa.h

```diff
@@ -15,22 +15,39 @@
 #define MLDSA65_KEYGEN_SEED_BYTES 32
 #define MLDSA65_SIGNATURE_SEED_BYTES 32
 
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
 int ml_dsa_65_keypair(uint8_t *public_key,
-                      uint8_t *private_key);
+                      uint8_t *secret_key);
+
+int ml_dsa_65_keypair_internal(uint8_t *public_key,
```

Comment:
We used `deterministic` for ML-KEM as we wrote these functions before the final release of the standard came out, and we were working from the IPD and naming conventions from the PQC mailing list. Since the release of FIPS 203, 204 final, we now see these deterministic methods described as `internal`, so to make the implementation of the standard clearer I named them internal and added descriptors to the standard itself, e.g.:

```
/*************************************************
 * Name:        crypto_sign_keypair_internal
 *
 * Description: FIPS 204: Algorithm 6 ML-DSA.KeyGen_internal.
 *              Generates public and private key. Internal API.
 ...


```
To refer to:
<img width="323" alt="Screenshot 2024-11-18 at 4 56 52 PM" src="https://github.com/user-attachments/assets/e891b081-0631-4ebf-8776-46a1f48c6d48">

Reply:
Do we have an issue tracking the renaming of ML-KEM functions?

Reply:
compiling little nits I come across here https://github.com/aws/aws-lc/issues/2002